app/services/migration/migrators/user.rb

```diff
@@ -30,21 +30,22 @@ def call
       migrate(self.class.ecf_users) do |ecf_user|
         user = find_or_initialize_user(ecf_user)
 
-        trns = unique_validated_trns(ecf_user)
-        validate_multiple_trns!(trns, user)
+        application_trns = validated_application_trns(ecf_user)
+        raise_if_multiple_application_trns_and_no_teacher_profile_trn!(application_trns, ecf_user)
+        trn = ecf_user.teacher_profile&.trn || application_trns.last || user&.trn
 
         npq_application = most_recent_created_npq_application(ecf_user)
         email = npq_application&.participant_identity&.email
 
         user.update!(
-          trn: ecf_user.teacher_profile&.trn ? ecf_user.teacher_profile.trn : trns.last,
+          trn:,
           full_name: ecf_user.full_name || user.full_name,
           email:,
           uid: ecf_user.get_an_identity_id || user.uid,
           date_of_birth: npq_application.date_of_birth || user.date_of_birth,
           national_insurance_number: npq_application.nino || user.national_insurance_number,
           active_alert: npq_application.active_alert || user.active_alert,
-          trn_verified: npq_application.teacher_reference_number_verified || user.trn_verified,
+          trn_verified: user&.trn_verified || trn.present?,
```

Comment:
Not sure about this, I think if there is a teacher profile with a trn we are sure that the trn is verified, if there is 1 trn we can read it from the verified on NPQ application from ECF, after that we can fallback to the one on the user. However as you implemented if there is no trn then this value should be false 👍